cloudformation/cw-dashboard-substack.cfn.yaml

```diff
@@ -0,0 +1,241 @@
+    - !Ref 'CreateFSXSubstack'
+    - 'true'
+  CreateRAIDSubstack: !Equals
+    - !Ref 'CreateRAIDSubstack'
+    - 'true'
+  # Duplicate of the raid-substack code
+  UseRaidVol1: !Not
+    - !Equals
+      - !Select
+        - '0'
+        - !Ref 'RAIDOptions'
+      - NONE
+  UseRaidVol2: !And
+    - !Not
+      - !Equals
+        - !Select
+          - '2'
+          - !Ref 'RAIDOptions'
+        - '1'
+    - !Condition 'UseRaidVol1'
+  UseRaidVol3: !And
+    - !Not
+      - !Equals
+        - !Select
+          - '2'
+          - !Ref 'RAIDOptions'
+        - '2'
+    - !Condition 'UseRaidVol2'
+  UseRaidVol4: !And
+    - !Not
+      - !Equals
+        - !Select
+          - '2'
+          - !Ref 'RAIDOptions'
+        - '3'
+    - !Condition 'UseRaidVol3'
+  UseRaidVol5: !And
+    - !Not
+      - !Equals
+        - !Select
+          - '2'
+          - !Ref 'RAIDOptions'
+        - '4'
+    - !Condition 'UseRaidVol4'
+  # Duplicate of the ebs-substack code
+  UseEBSVol2: !Not
+    - !Equals
+      - !Ref 'NumberOfEBSVol'
+      - '1'
+  UseEBSVol3: !And
+    - !Not
+      - !Equals
+        - !Ref 'NumberOfEBSVol'
+        - '2'
+    - !Condition 'UseEBSVol2'
+  UseEBSVol4: !And
+    - !Not
+      - !Equals
+        - !Ref 'NumberOfEBSVol'
+        - '3'
+    - !Condition 'UseEBSVol3'
+  UseEBSVol5: !And
+    - !Not
+      - !Equals
+        - !Ref 'NumberOfEBSVol'
+        - '4'
+    - !Condition 'UseEBSVol4'
+Outputs: {}
+Parameters:
+  CreateFSXSubstack:
+    Description: Parameter telling if a FSx Substack has been created
+    Type: String
+  FSXFileSystemId:
+    Description: ID of the FSx volume used
+    Type: String
+  RAIDOptions:
+    Description: Comma Separated List of RAID related options, 8 parameters in total, [shared_dir,raid_type,num_of_raid_volumes,volume_type,volume_size,volume_iops,encrypted,ebs_kms_key_id]
+    Type: CommaDelimitedList
+  RAIDVolumesIDs:
+    Description: Volume IDs of the resulted RAID EBS volumes
+    Type: CommaDelimitedList
+  CreateRAIDSubstack:
+    Description: Parameter telling if a RAID Substack has been created
+    Type: String
+  EBSVolumesIDs:
+    Description: IDs of the EBS volumes used
+    Type: CommaDelimitedList
+  NumberOfEBSVol:
+    Description: Number of EBS Volumes the user requested, up to 5
+    Type: Number
+  EFSFileSystemID:
+    Description: ID of the EFS volume used
+    Type: String
+  ClusterName:
+    Description: Name of the cluster to which this dashboard belongs
+    Type: String
+  CreateEFSSubstack:
+    Description: Parameter telling if an EFS Substack has been created
+    Type: String
+  MasterInstanceID:
+    Description: ID of the Master instance
+    Type: AWS::EC2::Instance::Id
+Resources:
+  MasterCloudWatchDashboard:
+    Type: AWS::CloudWatch::Dashboard
+    Properties:
+      DashboardName: !Join # Name like 'mycluster-Master'
+        - '-'
+        - - !Select
+            - '1'
+            - !Split
+              - 'parallelcluster-'
```

Comment:
It seems to me that you always need the cluster name and not the stack name so you can move the split in the main json template and only pass the cluster name to the substack template.

Reply:
We do not split anymore (but still make a join to add -Master). This is thus resolved I guess